backend-project/small_eod/institutions/tests/test_validators.py

```diff
@@ -81,3 +88,24 @@ def test_regon_length(self):
 
         form = f(data=dict(regon=chars_14))
         self.assertTrue(form.is_valid())
+
+
+class InstitutionValidatorsTestCase(TestCase):
+    def test_level_3(self):
+        """
+        Administrative unit must be a level 3 unit.
+        """
+        category1 = CategoryFactory(level=1)
+        administrative_unit = JednostkaAdministracyjnaFactory(category=category1)
+
+        f = modelform_factory(Institution, fields=("administrative_unit",))
+
+        self.assertEqual(administrative_unit.category.level, 1)
+        form = f(data=dict(administrative_unit=administrative_unit))
+        self.assertFalse(form.is_valid())
+
```

Comment:
Proponuje rozdzielić to na dwa testy. Zdaje się wystarczy napisać ```def test_....```. Dla przypadku pozytywnego i dla przypadku negatywnego.

Reply:
Zrobione